src/App.tsx

```diff
@@ -38,20 +38,23 @@ const HomePageRedirect = () => {
   const { data: currentUser } = useUser()
   const { data: internalUser } = useInternalUser({})
   const { params } = useLocationParams()
+  // @ts-expect-error useLocationParams needs to be typed
+  const { setup_action: setupAction, to } = params
 
   let redirectURL = '/login'
 
   if (internalUser && internalUser.owners) {
     redirectURL = `/${internalUser.owners[0]?.service}/${internalUser.owners[0]?.username}`
+    if (to === 'plan') {
+      redirectURL = '/plan' + redirectURL
```

Comment:
so this would be /plan/login? 🤔

Reply:
on 47 we're updating the redirectURL if we enter the if clause! /login is just the "base" value if we never hit any of the if statements

Reply:
This pattern should allow us to have links like app.codecov.io?to="_____" which can now utilize the user's session and direct them to where we need to go in the app

For now, I think we only have a single case for this, but if we decide to build out this pattern more it might make sense to have a config somewhere mapping the to values to a path

Reply:
this is cool